### ISSUE TITLE
vscode collection icons are rendered as *

### ISSUE THREAD

**Issue by noblekid96** [jdtsmith/kind-icon#42]
I recently upgraded kind-icons to 0.2.2 and svg-lib to 0.3

After upgrading the packages, I noticed an issue with vscode icons not being rendered properly as shown in the command output of `kind-icon-preview-all`

<img width="192" alt="image" src="https://github.com/jdtsmith/kind-icon/assets/44386386/4b66ae63-d6eb-41dc-b692-09fd36f84d88">

My kind-icon specific config is as follows:

```
(use-package! kind-icon
  :after corfu
  :custom
  (kind-icon-default-face 'corfu-default) ; to compute blended backgrounds correctly
  (kind-icon-mapping
        '((array          "a"   :icon "symbol-array"       :face font-lock-type-face              :collection "vscode")
          (boolean        "b"   :icon "symbol-boolean"     :face font-lock-builtin-face           :collection "vscode")
          (color          "#"   :icon "symbol-color"       :face success                          :collection "vscode")
          (command        "cm"  :icon "chevron-right"      :face default                          :collection "vscode")
          (constant       "co"  :icon "symbol-constant"    :face font-lock-constant-face          :collection "vscode")
          (class          "c"   :icon "symbol-class"       :face font-lock-type-face              :collection "vscode")
          (constructor    "cn"  :icon "symbol-method"      :face font-lock-function-name-face     :collection "vscode")
          (enum           "e"   :icon "symbol-enum"        :face font-lock-builtin-face           :collection "vscode")
          (enummember     "em"  :icon "symbol-enum-member" :face font-lock-builtin-face           :collection "vscode")
          (enum-member    "em"  :icon "symbol-enum-member" :face font-lock-builtin-face           :collection "vscode")
          (event          "ev"  :icon "symbol-event"       :face font-lock-warning-face           :collection "vscode")
          (field          "fd"  :icon "symbol-field"       :face font-lock-variable-name-face     :collection "vscode")
          (file           "f"   :icon "symbol-file"        :face font-lock-string-face            :collection "vscode")
          (folder         "d"   :icon "folder"             :face font-lock-doc-face               :collection "vscode")
          (function       "f"   :icon "symbol-method"      :face font-lock-function-name-face     :collection "vscode")
          (interface      "if"  :icon "symbol-interface"   :face font-lock-type-face              :collection "vscode")
          (keyword        "kw"  :icon "symbol-keyword"     :face font-lock-keyword-face           :collection "vscode")
          (macro          "mc"  :icon "lambda"             :face font-lock-keyword-face)
          (magic          "ma"  :icon "lightbulb-autofix"  :face font-lock-builtin-face           :collection "vscode")
          (method         "m"   :icon "symbol-method"      :face font-lock-function-name-face     :collection "vscode")
          (module         "{"   :icon "file-code-outline"  :face font-lock-preprocessor-face)
          (numeric        "nu"  :icon "symbol-numeric"     :face font-lock-builtin-face           :collection "vscode")
          (operator       "op"  :icon "symbol-operator"    :face font-lock-comment-delimiter-face :collection "vscode")
          (param          "pa"  :icon "gear"               :face default                          :collection "vscode")
          (property       "pr"  :icon "symbol-property"    :face font-lock-variable-name-face     :collection "vscode")
          (reference      "rf"  :icon "library"            :face font-lock-variable-name-face     :collection "vscode")
          (snippet        "S"   :icon "symbol-snippet"     :face font-lock-string-face            :collection "vscode")
          (string         "s"   :icon "symbol-string"      :face font-lock-string-face            :collection "vscode")
          (struct         "%"   :icon "symbol-structure"   :face font-lock-variable-name-face     :collection "vscode")
          (text           "tx"  :icon "symbol-key"         :face font-lock-doc-face               :collection "vscode")
          (typeparameter  "tp"  :icon "symbol-parameter"   :face font-lock-type-face              :collection "vscode")
          (type-parameter "tp"  :icon "symbol-parameter"   :face font-lock-type-face              :collection "vscode")
          (unit           "u"   :icon "symbol-ruler"       :face font-lock-constant-face          :collection "vscode")
          (value          "v"   :icon "symbol-enum"        :face font-lock-builtin-face           :collection "vscode")
          (variable       "va"  :icon "symbol-variable"    :face font-lock-variable-name-face     :collection "vscode")
          (t              "."   :icon "question"           :face font-lock-warning-face           :collection "vscode")))

  :config
  (add-to-list 'corfu-margin-formatters #'kind-icon-margin-formatter))
```

I have also checked that vscode icons are in the appropriate cache directory of svg-lib with the appropriate names

<img width="1563" alt="image" src="https://github.com/jdtsmith/kind-icon/assets/44386386/39db28b6-b32c-4679-a2d7-a286a149593f">

When I removed the vscode collection related configurations and used material icons, the icons were rendered properly.

<img width="167" alt="image" src="https://github.com/jdtsmith/kind-icon/assets/44386386/e3df3965-7c64-4ef6-8eb5-205883a30f4d">

Is there an issue with my configuration? Let me know if any further information is required

**Comment by jdtsmith** [comment]
Strange.  With the same combination using the [setup from the wiki](https://github.com/jdtsmith/kind-icon/wiki#using-vs-code-icons-as-an-alternative), I get:

<img width="215" alt="image" src="https://github.com/jdtsmith/kind-icon/assets/93749/bde0962d-ebeb-4a0b-9050-5674b32dbcd7">

You notice macro and module are the only two not taken from vscode-icons.  Can you clear out your vscode icons, restart and try again?

**Comment by danielpopescu** [comment]
have the same issue. macox, 29.1 native compilation, latest everything...

**Comment by jdtsmith** [comment]
Test from emacs -Q:
```elisp
(package-initialize)
(require 'kind-icon)

(setq kind-icon-mapping
      '((array "a" :icon "symbol-array" :face font-lock-type-face :collection "vscode")
	(boolean "b" :icon "symbol-boolean" :face font-lock-builtin-face :collection "vscode")
	(color "#" :icon "symbol-color" :face success :collection "vscode")
	(command "cm" :icon "chevron-right" :face default :collection "vscode")
	(constant "co" :icon "symbol-constant" :face font-lock-constant-face :collection "vscode")
	(class "c" :icon "symbol-class" :face font-lock-type-face :collection "vscode")
	(constructor "cn" :icon "symbol-method" :face font-lock-function-name-face :collection "vscode")
	(enum "e" :icon "symbol-enum" :face font-lock-builtin-face :collection "vscode")
	(enummember "em" :icon "symbol-enum-member" :face font-lock-builtin-face :collection "vscode")
	(enum-member "em" :icon "symbol-enum-member" :face font-lock-builtin-face :collection "vscode")
	(event "ev" :icon "symbol-event" :face font-lock-warning-face :collection "vscode")
	(field "fd" :icon "symbol-field" :face font-lock-variable-name-face :collection "vscode")
	(file "f" :icon "symbol-file" :face font-lock-string-face :collection "vscode")
	(folder "d" :icon "folder" :face font-lock-doc-face :collection "vscode")
	(function "f" :icon "symbol-method" :face font-lock-function-name-face :collection "vscode")
	(interface "if" :icon "symbol-interface" :face font-lock-type-face :collection "vscode")
	(keyword "kw" :icon "symbol-keyword" :face font-lock-keyword-face :collection "vscode")
	(macro "mc" :icon "lambda" :face font-lock-keyword-face)
	(magic "ma" :icon "lightbulb-autofix" :face font-lock-builtin-face :collection "vscode")
	(method "m" :icon "symbol-method" :face font-lock-function-name-face :collection "vscode")
	(module "{" :icon "file-code-outline" :face font-lock-preprocessor-face)
	(numeric "nu" :icon "symbol-numeric" :face font-lock-builtin-face :collection "vscode")
	(operator "op" :icon "symbol-operator" :face font-lock-comment-delimiter-face :collection "vscode")
	(param "pa" :icon "gear" :face default :collection "vscode")
	(property "pr" :icon "symbol-property" :face font-lock-variable-name-face :collection "vscode")
	(reference "rf" :icon "library" :face font-lock-variable-name-face :collection "vscode")
	(snippet "S" :icon "symbol-snippet" :face font-lock-string-face :collection "vscode")
	(string "s" :icon "symbol-string" :face font-lock-string-face :collection "vscode")
	(struct "%" :icon "symbol-structure" :face font-lock-variable-name-face :collection "vscode")
	(text "tx" :icon "symbol-key" :face font-lock-doc-face :collection "vscode")
	(typeparameter "tp" :icon "symbol-parameter" :face font-lock-type-face :collection "vscode")
	(type-parameter "tp" :icon "symbol-parameter" :face font-lock-type-face :collection "vscode")
	(unit "u" :icon "symbol-ruler" :face font-lock-constant-face :collection "vscode")
	(value "v" :icon "symbol-enum" :face font-lock-builtin-face :collection "vscode")
	(variable "va" :icon "symbol-variable" :face font-lock-variable-name-face :collection "vscode")
	(t "." :icon "question" :face font-lock-warning-face :collection "vscode")))
(kind-icon-preview-all)
```

Then `M-x eval-buffer`.  Then:

If this fails in the same way:

1.  Report on each of the precise versions of kind-icon and svg-lib.  
2. Delete everything in `.emacs.d/.cache/svg-lib` and restart `emacs -Q` and try the same again.  If this works try with your init.

If this does _not_ fail in the same way as with your init, it's a problem with your init.  Try adding likely things back in until failure.

**Comment by danielpopescu** [comment]
thanks. followed the instructions above. The only two icons showing up is macro and module. Everything else is *.

kind-icon: 0.2.2.0.20240106.145829
svg-lib: 0.3.0.20240104.62538

emacs 29.1. native comp. macos.

**Comment by jdtsmith** [comment]
To double check, this is with `emacs -Q`?  Try this simply, removing any mention of `kind-icon`:

```elisp
(package-initialize)
(require 'svg-lib)
(insert-image (svg-lib-icon "symbol-event"
			    '( :collection "vscode"
			       :face font-lock-variable-name-face
			       :padding 0 :stroke 0 :margin 0 :radius 0
			       :height 1.0 :scale 1.0 :background nil)))
```

You should see:

<img width="543" alt="image" src="https://github.com/jdtsmith/kind-icon/assets/93749/0e72152d-4460-4047-9204-f201ce2c596e">

**Comment by danielpopescu** [comment]
yes, emacs-Q. I tried the above. See nothing in the buffer. I do see that it generates .cache/svg-lib/vscode_symbol-event.svg but it is not dispalyed...

**Comment by jdtsmith** [comment]
OK in this case the problem is either `svg-lib` or your emacs' ability to read that SVG image.  `kind-icon` is not involved.  Please open an issue on `svg-lib` and link back here.  

You could also open that cached image in the browser to make sure it looks OK, and look at the SVG code `svg-lib` generates to see if anything looks suspicious, like this:

```elisp
(insert "\n"
 (pp-to-string
  (svg-lib-icon "symbol-event"
		'( :collection "vscode"
		   :face font-lock-variable-name-face
		   :padding 0 :stroke 0 :margin 0 :radius 0
		   :height 1.0 :scale 1.0 :background nil))))
(image :type svg :data "<svg width=\"14\" height=\"14.0\" version=\"1.1\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\"> <path transform=\"translate(0.000000,0.000000) scale(0.875000) translate(0.000000,0.000000)\" fill=\"#ffffff\" d=\"M7.41379 1.55996L8.31177 1H11.6059L12.4243 2.57465L10.2358 6H12.0176L12.7365 7.69512L5.61967 15L4.01699 13.837L6.11967 10H4.89822L4.00024 8.55996L7.41379 1.55996ZM7.78058 9L4.90078 14.3049L12.0176 7H8.31177L11.6059 2H8.31177L4.89822 9H7.78058Z\"></path></svg>" :scale 1 :ascent center :transform-smoothing t)
```

**Comment by jdtsmith** [comment]
I can confirm versions _more recent_ than the tagged version of svg-lib on elpa (v0.3) shows this problem.  I always recommend sticking with elpa stable and not elpa devel for packages, unless you are actively helping develop a package.

**Comment by noblekid96** [comment]
This issue is fixed after a fix from svg-lib rougier/svg-lib#43

https://github.com/rougier/svg-lib/commit/3c2724e37b1df448304d9bece4ec35282f6b363c
